src/library/dao/playlistdao.cpp

```diff
@@ -246,6 +246,29 @@ bool PlaylistDAO::isPlaylistLocked(const int playlistId) const {
     return false;
 }
 
+bool PlaylistDAO::clearPlaylist(const int playlistId) {
+    int position =
+        (m_pAutoDJProcessor && m_pAutoDJProcessor->nextTrackLoaded()) ? 2 : 1;
```

Comment:
Hm, though this is a general "clearPlaylist" function so it shouldn't have AutoDJ logic built in. Maybe a better name would be "removeTracksFromPlaylist(int startIndex)" and then you would pass in the appropriate start index from sendToAutoDJ as the add-to-top handling works.